cmd/nodeadm/debug/debug.go

```diff
@@ -95,40 +95,41 @@ func (c *debug) Run(log *zap.Logger, opts *cli.GlobalOptions) error {
 	runner := validation.NewRunner[*api.NodeConfig](printer)
 	apiServerValidator := node.NewAPIServerValidator(kubelet.New())
 	clusterProvider := kubernetes.NewClusterProvider(awsConfig)
+
+	// Register validations that do not require cluster details first
+	runner.Register(creds.Validations(awsConfig, nodeConfig)...)
+	runner.Register(
+		validation.New("ntp-sync", system.NewNTPValidator().Run),
+		validation.New("swap", system.NewSwapValidator().Run),
+		validation.New("ulimit", system.NewUlimitValidator().Run),
+		validation.New("aws-auth", sts.NewAuthenticationValidator(awsConfig).Run),
+		validation.New("proxy-config", network.NewProxyValidator().Run),
+	)
+
 	clusterDetail, err := clusterProvider.ReadClusterDetails(ctx, nodeConfig)
 	if err != nil {
 		// Only if reading the EKS fail is when we "start" a validation and signal it as failed.
 		// Otherwise, there is no need to surface we are reading from the EKS API.
-		printer.Starting(ctx, "kubernetes-endpoint-access", "Validating access to Kubernetes API endpoint")
-		printer.Done(ctx, "kubernetes-endpoint-access", err)
-		return err
-	}
-	cluster, err := eks.ReadCluster(ctx, awsConfig, nodeConfig)
-	if err != nil {
-		// Only if reading the EKS fail is when we "start" a validation and signal it as failed.
-		// Otherwise, there is no need to surface we are reading from the EKS API.
+		err = validation.WithRemediation(err, "Ensure the Kubernetes API server endpoint is provideed or "+
+			"the node has access and permissions to call EKS DescribeCluster API.")
 		printer.Starting(ctx, "kubernetes-endpoint-access", "Validating access to Kubernetes API endpoint")
```

Comment:
Unrelated to this change - let's rename this validation, since it's not validating access to Kubernetes API server endpoint